### ISSUE TITLE
Artifact downloads limited by max-jobs

### ISSUE THREAD

**Issue by bendlas** [jlesquembre/clj-nix#73]
When the jar and pom files for a classpath are not in /nix/store, the resulting downloads are treated as regular builds and can be very slow if `max-jobs` is low.

This is essentially the same problem https://github.com/nix-community/yarn2nix/issues/9#issuecomment-437074958

Possible workarounds
- A command that pre-builds just the classpath and adds `--option max-jobs 256`
- Maybe `builtins.fetchurl` wouldcount against `http-downloads`?

**Comment by jlesquembre** [comment]
> A command that pre-builds just the classpath and adds `--option max-jobs 256`

That's for sure an option. Is your use case to speed up CI builds? 


If not on CI, there is an option to add the dependencies to the nix store when you generate the lock file, search for `CLJNIX_ADD_NIX_STORE` in the README: 

```
CLJNIX_ADD_NIX_STORE=true nix run github:jlesquembre/clj-nix#deps-lock
```

Probably I should add a flag to the CLI to do that.

**Comment by bendlas** [comment]
> That's for sure an option. Is your use case to speed up CI builds?

Yes, also

- first-time user: time to command
- existing user: during/after version updates

**Comment by jlesquembre** [comment]
@bendlas to clarify it, a new CLI command, like 

```bash
add-clj-deps deps-lock.json
```

which adds all the dependencies to the nix store, solves your use case, right?

**Comment by bendlas** [comment]
Yes, a cli command like `preload-deps` to fast-track all the dependency downloads would solve my use case neatly.
